server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -257,12 +269,12 @@ private static HitContext prepareNonNestedHitContext(
 
         String id = idLoader.getId(subDocId);
         if (id == null) {
-            // TODO: can we use pooled buffers here as well?
-            SearchHit hit = SearchHit.unpooled(docId, null);
+            SearchHit hit = new SearchHit(docId);
+            // TODO: can we use real pooled buffers here as well?
```

Comment:
Moved the todo here now that we have the `SearchHit` side of pooling covered here, would be nice to eventually  make use of actual pooled buffers on the data nodes during a follow-up.

Reply:
Perhaps a silly question, but what's left that's unpooled here? (all `unpooled` calls have been removed in this method I think) The `TODO` is a bit confusing for me.

Reply:
Not silly at all, the actual bytes underlying the `SearchHit` source will still be a freshly allocated `byte[]` in some of these code paths. Would be nicer if we were copying from Lucene to our buffer pools :)
Right now the ref-counting here is not really managing any resources otherwise, except for in the case with inner hits that we're dealing with here :)